diff --git a/MODULE.bazel b/MODULE.bazel
index 8733088ece..c531d865aa 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -174,8 +174,8 @@ pip.parse(
         "numpy": ["numpy_headers"],
     },
     hub_name = "xla_pypi",
-    python_version = "3.11",
-    requirements_lock = "//:requirements_lock_3_11.txt",
+    python_version = "3.12",
+    requirements_lock = "//:requirements_lock_3_12.txt",
     whl_modifications = {
         "@pypi_mods//:numpy.json": "numpy",
     },
